transgene.py

```diff
@@ -526,41 +482,62 @@ def parse_peptides(infile, outfile):
     identical (to correct for splicing information).
     """
     with open(infile, 'r') as i_f:
-        peptides = collections.Counter()
+        peptides = collections.defaultdict(list)
         for pep_name, _, pep_seq in read_fasta(i_f, 'ARNDCQEGHILKMFPSTWYVBZJUOX'):
-            pep_name = pep_name.split('_')
-            gene_name = pep_name[0]
-            hugo_gene = pep_name[2]
-            transcript_mutation = '_'.join([pep_name[1]]+pep_name[3:])
-            if peptides[(gene_name, hugo_gene)] == 0:
-                peptides[(gene_name, hugo_gene)] = {transcript_mutation: pep_seq}
-            else:
-                peptides[(gene_name, hugo_gene)][transcript_mutation] = pep_seq
+            peptides[pep_seq].append(pep_name)
+
     outmap = {}
     with open(outfile, 'w') as o_f:
         # Mhc predictors can't handle the giant peptide names created by merging the names of
         # peptide emerging from transcripts bearing similar mutations hence we use an incremental
         # system to name the peptides and store the peptide number to actual peptide name
         # information in a map file.
         peptide_number = 1
-        for gene in peptides.keys():
-            unique_seqs = set(peptides[gene].values())
-            for group in [[x for x, y in peptides[gene].items() if y == z] for z in unique_seqs]:
-                pepname = ''.join(['neoepitope_', str(peptide_number)])
-                group_info = '\t'.join(list(gene) + [','.join(group)])
-                group_seq = peptides[gene][group[0]]
-                # Print to faa files
-                print('>', pepname, sep='', file=o_f)
-                print(group_seq, file=o_f)
-                # Save to map dict
-                outmap[pepname] = group_info
-                peptide_number += 1
+        for peptide in peptides.keys():
+            # Print to faa files
+            neoepitope_id = 'neoepitope_%d' % peptide_number
+            print('>%s\n%s' % (neoepitope_id, peptide), file=o_f)
+
+            # Save mapping of neoepitope identifier to mutation information
+            outmap[neoepitope_id] = '\t'.join(peptides[peptide])
```

Comment:
Will this be the same as the info that was dumped in the previous map? Remember, rankboost uses the map info to fill in some fields and if this changes things, we need to modify that as well.